sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py

```diff
@@ -282,11 +282,14 @@ def _get_effective_partition_key_for_multi_hash_partitioning_v2(
     def _is_prefix_partition_key(
             self,
             partition_key: Union[str, int, float, bool, Sequence[Union[str, int, float, bool, None]], Type[NonePartitionKeyValue]]) -> bool: # pylint: disable=line-too-long
-        if self.kind!= "MultiHash":
+        if self.kind != "MultiHash":
             return False
-        if isinstance(partition_key, list) and len(self['paths']) == len(partition_key):
+        is_tuple_or_list = True in (isinstance(partition_key, list),
+                                    isinstance(partition_key, tuple))
+        if is_tuple_or_list and len(self['paths']) == len(partition_key):
             return False
-        return True
+
+        return is_tuple_or_list
```

Comment:
should it just be return true? I guess a quick question here -> for prefix partition, can customer just passing Union[str, int, float, bool] if they only want to pass the first component of the partition key?

Reply:
The logic flow of getting the epk expects a sequence like a list and a tuple. Passing in a string, or any of the mentioned values would require a rework of it so it doesn't error out (ie trying to iterate an int), but it would be better to just have the user encase that value in a list rather than add extra logic to accommodate that use case. We should only allow lists and tuples otherwise we let the backend give us a 400 bad request error indicating the pk value is not what was expected.